main.go

```diff
@@ -1,22 +1,22 @@
-package dnsserv
+package main
 
 import (
-	"github.com/twtiger/toy-dns-nameserver/reqhandler"
-	"net"
+	"fmt"
+	"log"
 )
 
-func main() {
+func initLogger() {
+	flags := log.Ldate | log.Ltime | log.Llongfile
+	log.SetFlags(flags)
+	log.SetPrefix("[logger:] ")
+}
 
-	udpAddr := &net.UDPAddr{
-		IP:   net.ParseIP("127.0.0.1"),
-		Port: 53,
-	}
+func main() {
+	initLogger()
 
-	for {
-		udpConn, err := net.ListenUDP("udp", udpAddr)
-		if err != nil {
-			// handle error
-		}
-		go reqhandler.HandleUDPConnection(udpConn)
+	err := run()
```

Comment:
run is scoped to the main package, we don't need to export it. 

We can think over what should be in a separate package and what belongs in main.